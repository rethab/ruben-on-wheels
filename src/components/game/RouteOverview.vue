<template>
  <div>
    <v-timeline side="end" density="compact">
      <v-timeline-item
        :dot-color="dotColor(city.name)"
        size="x-small"
        v-for="(city, index) in cities"
        :key="index"
      >
        <v-card density="compact">
          <v-card-subtitle>{{ city.name }}</v-card-subtitle>
          <v-card-text>
            {{ showPlayers(city.name) }}
          </v-card-text>
        </v-card>
      </v-timeline-item>
    </v-timeline>
  </div>
</template>

<script lang="ts">
import { useGameStore } from "@/stores/game";
import { cities } from "@/services/cities";

export default {
  setup() {
    return { cities };
  },
  methods: {
    dotColor(cityName: string) {
      const { currentPlayer } = useGameStore();
      const color = currentPlayer?.currentCity === cityName ? "orange" : "grey";
      return `${color}-lighten-2`;
    },
    showPlayers(name: string) {
      const { playersInCity } = useGameStore();
      const players = playersInCity(name);

      if (players.length === 0) return;

      return players.map(({ name }) => name).join(", ");
    },
  },
};
</script>
