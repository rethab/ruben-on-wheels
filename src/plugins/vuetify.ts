import "vuetify/styles";
import { createVuetify } from "vuetify";
import { aliases, mdi } from "vuetify/iconsets/mdi";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";
import "@mdi/font/css/materialdesignicons.css";

export default createVuetify({
  icons: { defaultSet: "mdi", aliases, sets: { mdi } },
  components,
  directives,
});

// Export for test.
export { components, directives };
