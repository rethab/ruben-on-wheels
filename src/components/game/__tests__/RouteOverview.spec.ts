import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import RouteOverview from "../RouteOverview.vue";

import vuetify from "@/plugins/vuetify";

function mountComponent() {
  return mount(RouteOverview as any, {
    props: {
      players: [
        { name: "Alice", currentCity: "Zurich" },
        { name: "Bob", currentCity: "Basel" },
        { name: "Carl", currentCity: "Basel" },
      ],
      currentPlayer: {
        name: "Bob",
        currentCity: "Basel",
      },
      cities: [{ name: "Zurich" }, { name: "Basel" }, { name: "Köln" }],
    },
    global: {
      plugins: [vuetify],
    },
  });
}

describe("RouteOverview", () => {
  it("renders full component", () => {
    const wrapper = mountComponent();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("players are shown in their city", () => {
    const wrapper = mountComponent();
    const [zurich, basel, cologne] = wrapper.findAll(".v-timeline-item");

    expect(zurich.find(".v-card-subtitle").text()).toBe("Zurich");
    expect(zurich.find(".v-card-text").text()).toBe("Alice");

    expect(basel.find(".v-card-subtitle").text()).toBe("Basel");
    expect(basel.find(".v-card-text").text()).toBe("Bob, Carl");

    expect(cologne.find(".v-card-subtitle").text()).toBe("Köln");
    expect(cologne.find(".v-card-text").text()).toBe("");
  });

  it("city of current player is highlighted", () => {
    const wrapper = mountComponent();

    const [zurich, basel, cologne] = wrapper.findAll(".v-timeline-item");
    expect(
      zurich
        .find(".v-timeline-divider__dot")
        .find(".bg-orange-lighten-2")
        .exists()
    ).toBeFalsy();
    expect(
      basel
        .find(".v-timeline-divider__dot")
        .find(".bg-orange-lighten-2")
        .exists()
    ).toBeTruthy();
    expect(
      cologne
        .find(".v-timeline-divider__dot")
        .find(".bg-orange-lighten-2")
        .exists()
    ).toBeFalsy();
  });
});
