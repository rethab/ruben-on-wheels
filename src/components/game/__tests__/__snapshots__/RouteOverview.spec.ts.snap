// Vitest Snapshot v1

exports[`RouteOverview > renders full component 1`] = `
"<div>
  <div class=\\"v-timeline v-timeline--vertical v-timeline--align-center v-timeline--justify-auto v-theme--light v-timeline--density-compact v-timeline--side-end\\" style=\\"--v-timeline-line-thickness: 2px;\\">
    <div class=\\"v-timeline-item\\" style=\\"--v-timeline-dot-size: 0px; --v-timeline-line-inset: 0px;\\">
      <div class=\\"v-timeline-item__body\\">
        <div class=\\"v-card v-theme--light v-card--density-compact v-card--variant-elevated\\">
          <!---->
          <div class=\\"v-card__loader\\">
            <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <!---->
          <div class=\\"v-card-subtitle\\">Zurich</div>
          <div class=\\"v-card-text\\">Alice</div>
          <!---->
          <!----><span class=\\"v-card__underlay\\"></span>
        </div>
      </div>
      <div class=\\"v-timeline-divider\\">
        <div class=\\"v-timeline-divider__before\\"></div>
        <div class=\\"v-timeline-divider__dot v-timeline-divider__dot--size-x-small\\">
          <div class=\\"v-timeline-divider__inner-dot bg-grey-lighten-2\\">
            <!---->
          </div>
        </div>
        <div class=\\"v-timeline-divider__after\\"></div>
      </div>
      <!---->
    </div>
    <div class=\\"v-timeline-item\\" style=\\"--v-timeline-dot-size: 0px; --v-timeline-line-inset: 0px;\\">
      <div class=\\"v-timeline-item__body\\">
        <div class=\\"v-card v-theme--light v-card--density-compact v-card--variant-elevated\\">
          <!---->
          <div class=\\"v-card__loader\\">
            <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <!---->
          <div class=\\"v-card-subtitle\\">Basel</div>
          <div class=\\"v-card-text\\">Bob, Carl</div>
          <!---->
          <!----><span class=\\"v-card__underlay\\"></span>
        </div>
      </div>
      <div class=\\"v-timeline-divider\\">
        <div class=\\"v-timeline-divider__before\\"></div>
        <div class=\\"v-timeline-divider__dot v-timeline-divider__dot--size-x-small\\">
          <div class=\\"v-timeline-divider__inner-dot bg-orange-lighten-2\\">
            <!---->
          </div>
        </div>
        <div class=\\"v-timeline-divider__after\\"></div>
      </div>
      <!---->
    </div>
    <div class=\\"v-timeline-item\\" style=\\"--v-timeline-dot-size: 0px; --v-timeline-line-inset: 0px;\\">
      <div class=\\"v-timeline-item__body\\">
        <div class=\\"v-card v-theme--light v-card--density-compact v-card--variant-elevated\\">
          <!---->
          <div class=\\"v-card__loader\\">
            <div class=\\"v-progress-linear v-theme--light\\" style=\\"height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <!---->
          <div class=\\"v-card-subtitle\\">KÃ¶ln</div>
          <div class=\\"v-card-text\\"></div>
          <!---->
          <!----><span class=\\"v-card__underlay\\"></span>
        </div>
      </div>
      <div class=\\"v-timeline-divider\\">
        <div class=\\"v-timeline-divider__before\\"></div>
        <div class=\\"v-timeline-divider__dot v-timeline-divider__dot--size-x-small\\">
          <div class=\\"v-timeline-divider__inner-dot bg-grey-lighten-2\\">
            <!---->
          </div>
        </div>
        <div class=\\"v-timeline-divider__after\\"></div>
      </div>
      <!---->
    </div>
  </div>
</div>"
`;
